app/gui/src/project-view/stores/rightDock.ts

```diff
@@ -0,0 +1,117 @@
+import { createContextStore } from '@/providers'
+import { computedFallback } from '@/util/reactivity'
+import { computed, proxyRefs, ref, toRef } from 'vue'
+import { assertNever } from 'ydoc-shared/util/assert'
+import { unwrapOr } from 'ydoc-shared/util/data/result'
+import { GraphStore } from './graph'
+import { PersistedStore } from './persisted'
+import { useSettings } from './settings'
+
+export type RightDockStore = ReturnType<typeof useRightDock>
+
+export enum StorageMode {
+  Default,
+  ComponentBrowser,
+}
+
+export const [provideRightDock, useRightDock] = createContextStore(
+  'rightDock',
+  (graph: GraphStore, persisted: PersistedStore) => {
+    const inspectedAst = computed(() => unwrapOr(graph.methodAst, undefined))
+    const { user: userSettings } = useSettings()
+
+    const storageMode = ref(StorageMode.Default)
+    const markdownDocs = computed(() => inspectedAst.value?.mutableDocumentationMarkdown())
+
+    const defaultVisible = computedFallback(
+      toRef(persisted, 'graphRightDock'),
+      () => (markdownDocs.value?.length ?? 0) > 0,
+    )
+    const defaultTab = computedFallback(toRef(persisted, 'graphRightDockTab'), () => 'docs')
+
+    const width = toRef(persisted, 'graphRightDockWidth')
+
+    const cbVisible = ref(true)
+    const cbTab = ref('help')
```

Comment:
The tab names seem to be magic strings, which risks bugs if we change the available tabs in the future.